modelon/impact/client/client.py

```diff
@@ -71,7 +72,16 @@ def _authenticate_against_api(self, interactive):
         else:
             login_data = {"secretKey": self.api_key}
 
-        self._sal.api_login(login_data)
+        try:
+            self._sal.api_login(login_data)
+        except modelon.impact.client.sal.exceptions.HTTPError as error:
+            if interactive:
+                self._credentials.delete_key()
```

Comment:
It does not seem ok to me deleting the users key if something goes wrong. Could be very destructive if this is not what the user wants (especially if the only place they keep the key is in the file).

It might be a good idea to move this one level up:
```
try:
  self._authenticate_against_api(interactive)
except modelon.impact.client.sal.exceptions.HTTPError as error:
  if interactive:
    logger.warning('Could not login with configured API key, re-trying with input from prompt')
    self.api_key = self._credentials.get_key_from_prompt()
    self._authenticate_against_api(interactive)
  else:
    raise
```